pkg/trace/agent/run.go

```diff
@@ -39,9 +39,14 @@ func Run(ctx context.Context) {
 	}
 
 	cfg, err := config.Load(flags.ConfigPath)
+	//initializing logger before checking the error to ensure start-up errors are logged to the log file
+	//even if cfg is loaded with an error or does not exist, valid log file will still be created
+	initLogger(cfg)
```

Comment:
I don't think there's any point to create a dedicated function for just a single function call. We didn't have it before so why have it now? In my opinion it makes sense to have a dedicated function if these conditions are met:
* The code in the parent function is complex and/or confusing
* The new function has a concise purpose
* The function is reused
In this case, only the second one is true.

Furthermore, when reading through this function and reaching a call to `coreconfig.SetupLogger` one must not travel to a different line to understand another concept.

We should be wary of introducing as little new concept as needed (referring to `initLogger` here).